_adr/82/index.adoc

```diff
@@ -0,0 +1,87 @@
+---
+num: 82
+title: Streaming Unit Hour Metrics
+status: "Draft"
+authors:
+  - "Peter Braun"
+tags:
+  - "kafka"
+#applies_padrs: # What PADRs does this ADR apply?
+#applies_patterns: # What APs does this ADR apply?
+---
+
+## Context and problem statement
+
+Streaming Unit pricing will be accomplished by introducing a new concept called ‘Streaming Unit Hours’. This is defined as Cluster Hours multiplied by the Streaming Unit Size. Here Streaming Unit Size should not be confused with Quota usage which is required for prepaid pricing.
+
+Cluster Hours are currently recorded on the data plane clusters using a recording rule (kafka_id:strimzi_resource_state:max_over_time1h). We need a way to multiply the value of this with the Streaming Unit Size for a given instance.
+
+Other metrics don’t need to be multiplied: storage and traffic are billed independently of the Streaming Unit.
+
+## Goals
+
+* Modify the recording rule kafka_id:strimzi_resource_state:max_over_time1h to multiply its value by the Streaming Unit Size.
+
+## Non-goals
+
+* Changes on the Subwatch side. This ADR is only concerned with the Managed Kafka changes.
+* Define additional fields to hold the cost per Streaming Unit hour. The amount is simply multiplied by the instance size.
+
+Stakeholders
+
+* Eng
+* BU
+* QE
+
+## Current architecture
+
+Cluster Hours are defined in our link:https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk[Observability Resources] repository among other consumption related metrics. It is collected on the data plane and remote written to Observatorium. From there it is made available to Subscription Watch who use it to create invoices.
+
+Cluster Hours is based on the maximum-per-one-hour value of the underlying strimzi_resource_state metric which is 1 when a Kafka CR is present in the namespace. Thus, our billing is accurate to the hour.
+
+The Kafka instance size in Streaming Units is not taken into account at any point here.
+
+## Proposed architecture
+
+Proposed Architecture
+
+*Firstly*, the size in Streaming Units needs to be available to the data plane. That is currently not the case. The options to achieve this are:
+
+1. Include it in the Managed Kafka CR as an annotation. The CR is retrieved by the Fleetshard sync component via the /api/kafkas_mgmt/v1/agent-clusters/{id}/kafkas endpoint.
+2. Include it in the Managed Kafka CR Capacity spec. Also retrieved by the Fleetshard sync component via the /api/kafkas_mgmt/v1/agent-clusters/{id}/kafkas endpoint.
```

Comment:
Won't including it as an annotation suggested above suffice?

Reply:
Yes, that should be fine. I just wanted to list all options but we can say that the annotation is preferred.

Reply:
Thanks.